syntax = "proto3";


message Empty {}

message StatusResponse {
  bool status = 1;
  string statusMessage = 2;
}

message FileOpenRequest {
  string fileName = 1;
//   string hdu = 2;
//   string uuid = 3;
}

message FileCloseRequest {
  string uuid = 2;
}

message FileInfoRequest {
  string uuid = 1;
}

message FileInfoResponse {
  int32 hduNum = 2;
  int32 hduType = 3;
  int32 dataType = 4;
  repeated int32 hduShape = 5;
}

// message ImageDataRequest {
//   string uuid = 1;
//   repeated int32 start = 2;
//   int64 numPixels = 3;
// }

// message ImageDataResponse {
//   bytes data = 1;
// }

// message SpectralProfileRequest {
//   string uuid = 1;
//   int32 x = 2;
//   int32 y = 3;
//   int32 z = 4;
//   int32 numPixels = 5;
//   int32 width = 6;
//   int32 height = 7;
// }

// message SpectralProfileResponse {
//   bytes data = 1;
// }

service H5ReaderService{
  rpc CheckStatus(Empty) returns  (StatusResponse) {}
  rpc OpenFile(FileOpenRequest) returns (StatusResponse) {}
  rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
  rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}

//   rpc GetImageData (ImageDataRequest) returns (ImageDataResponse) {}
//   rpc GetSpectralProfile (SpectralProfileRequest) returns (SpectralProfileResponse) {}

}